cvat/apps/engine/signals.py

```diff
@@ -36,13 +37,28 @@ def __save_job_handler(instance, created, **kwargs):
         db_task.status = status
         db_task.save(update_fields=["status", "updated_date"])
 
-@receiver(post_save, sender=User,
-    dispatch_uid=__name__ + ".save_user_handler")
-def __save_user_handler(instance, **kwargs):
+
+@receiver(post_save, sender=User, dispatch_uid=__name__ + ".save_user_handler")
+def __save_user_handler(instance: User, **kwargs):
+    should_access_analytics = instance.is_superuser or instance.groups.filter(name=settings.IAM_ADMIN_ROLE).exists()
     if not hasattr(instance, 'profile'):
         profile = Profile()
         profile.user = instance
+        profile.has_analytics_access = should_access_analytics
         profile.save()
+    elif should_access_analytics and not instance.profile.has_analytics_access:
+        instance.profile.has_analytics_access = True
+        instance.profile.save()
+
+
+@receiver(m2m_changed, sender=User.groups.through, dispatch_uid=__name__ + ".m2m_user_groups_change_handler")
+def __m2m_user_groups_change_handler(sender, instance: User, action: str, **kwargs):
+    if action == 'post_add':
```

Comment:
Do we need to handle also `post_remove`?

Reply:
Its possible to give access to analytics to not-admin users, therefore I think analytics access should not be revoked with admin group

Reply:
I meant handling cases when a user had the `admin` group and then he was excluded from this group.
Could be discussed and fixed later I guess.